src/libsync/vfs/cfapi/cfapiwrapper.cpp

```diff
@@ -346,7 +467,7 @@ OCC::Result<OCC::CfApiWrapper::ConnectionKey, QString> OCC::CfApiWrapper::connec
     if (result != S_OK) {
         return QString::fromWCharArray(_com_error(result).ErrorMessage());
     } else {
-        return key;
+        return std::move(key);
```

Comment:
Dont use std::move on returns, it has no effect (I oversimplify a bit, it's often worse than that).

Reply:
> Dont use std::move on returns, it has no effect (I oversimplify a bit, it's often worse than that).

@er-vin Well, it gives a warning here every time I build the project. So, I decided to fix it. Now - there is no warning anymore, and the code still works. So, I guess - it's a win? :) Or, no?

Reply:
Hm, odd. Which warning is that? Sounds like a false positive. I seem to remember Visual Studio has kind of an old bug regarding this but I can't remember the specifics. Thing is that can prevent some return optimizations to use std::move on returns.

Reply:
> Hm, odd. Which warning is that? Sounds like a false positive. I seem to remember Visual Studio has kind of an old bug regarding this but I can't remember the specifics. Thing is that can prevent some return optimizations to use std::move on returns.

@er-vin This is coming from the Qt Creator:
`cfapiwrapper.cpp:493:16: error: call to implicitly-deleted copy constructor of 'OCC::CfApiWrapper::ConnectionKey'
cfapiwrapper.h:38:44: note: copy constructor of 'ConnectionKey' is implicitly deleted because field '_data' has a deleted copy constructor
memory:2551:5: note: 'unique_ptr' has been explicitly marked deleted here
result.h:34:14: note: passing argument to parameter 'value' here`

Reply:
Ah OK. Got it now. This is because of the type elision. I think the better fix is: `return { std::move(key) };`

This should result in the same generated code but would avoid the confusion on read that the move is not being for what's returned. This is clearly what tripped me up here. :-)